Take the rustc target from RUST_TARGET instead of matching the C
triple against rustc's target list.

--- a/build/moz.configure/rust.configure	2026-09-28 03:08:00.285308199 +0530
+++ b/build/moz.configure/rust.configure	2026-09-28 03:08:00.298116979 +0530
@@ -278,6 +278,7 @@
     return data
 
 
+@imports("os")
 def detect_rustc_target(host_or_target, arm_target, rust_supported_targets):
     # Rust's --target options are similar to, but not exactly the same
     # as, the autoconf-derived targets we use.  An example would be that
@@ -441,7 +442,7 @@
 
         return None
 
-    rustc_target = find_candidate(candidates)
+    rustc_target = os.environ["RUST_TARGET"]
 
     if rustc_target is None:
         die("Don't know how to translate {} for rustc".format(host_or_target.alias))
